haskell/private/actions/link.bzl

```diff
@@ -352,15 +352,10 @@ def link_library_dynamic(hs, cc, dep_info, cc_info, extra_srcs, objects_dir, my_
     )
     args.add_all(pkg_info_args)
 
-    # When linking a dynamic library we still collect static libraries for
-    # dependencies where possible. This is so that a final binary that depends
-    # on this dynamic library, is linked statically itself, will not fail at
-    # link time due to missing transitive dynamic library dependencies. In this
-    # case transitive dependencies will still be linked in statically.
```

Comment:
GHC loads shared libraries with `RTLD_LAZY`. This can cause missing symbols with transitive Haskell library dependencies in dynamic linking mode. This issue also came up in https://github.com/tweag/rules_haskell/pull/927. It's not strictly related to the `cc_wrapper`. However, I tested the `cc_wrapper` on `damlc` in https://github.com/digital-asset/daml with a dynamic RTS, and that required this change to pass.